src/main/java/org/wahlzeit_revisited/repository/PhotoRepository.java

```diff
@@ -0,0 +1,143 @@
+package org.wahlzeit_revisited.repository;
+
+import jakarta.inject.Inject;
+import org.wahlzeit_revisited.model.Photo;
+import org.wahlzeit_revisited.model.PhotoFactory;
+import org.wahlzeit_revisited.model.PhotoStatus;
+import org.wahlzeit_revisited.model.User;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class PhotoRepository extends AbstractRepository<Photo> {
+
+    @Inject
+    public PhotoFactory factory;
+
+    /*
+     * AbstractRepository contract
+     */
+
+    @Override
+    protected Optional<Photo> doFindById(Long id) throws SQLException {
```

Comment:
I usually move the primitive methods to the end of the file. Main structure is by role, within a role first public, then protected, then private. Shared protected or private stuff to the end. I don't like variables at the top but since this is the accepted pattern I roll with it.

Reply:
changed in: 224ed9ba579f1d0a0922a953558005c9a280497f